ui/src/pages/Create.js

```diff
@@ -0,0 +1,645 @@
+    // to assign them IDs.
+    this.profileFieldIdCounter = 0;
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.addProfilePageField = this.addProfilePageField.bind(this);
+    this.removeProfilePageField = this.removeProfilePageField.bind(this);
+    this.photoUploadInput = React.createRef();
+  }
+
+  componentDidMount() {
+    const apiUrl = '/' + this.repoId + '/d/repo';
+    fetch(apiUrl)
+      .then(res => res.json())
+      .then(
+        (repo) => {
+          this.setState({
+            isLoaded: true,
+            repo: repo,
+          });
+        },
+        (error) => {
+          this.setState({
+            isLoaded: true,
+            error: error
+          });
+        }
+      );
+  }
+
+  renderMoreFieldsButton() {
+    return (
+      <div className='create-formgroupwrapper create-morebutton'>
+        <span
+          className='mdc-typography--body1'
+          onClick={(e) => this.setState({showAllIdInfoFields: true})}
+        >
+          <FormattedHTMLMessage {...messages.moreDropdown} />
+        </span>
+      </div>
+    )
+  }
+
+  renderPhotoFields() {
+    var photoFilenameSpan = this.state.formPhotoFile
+        ? <span>{this.state.formPhotoFile.name}</span>
+        : null;
+    var photoDeleteButton = null;
+    if (this.state.formPhotoFile) {
+      photoDeleteButton = (
+        <span onClick={(e) => {this.setState({formPhotoFile: ''})}}>X</span>
+      );
+    }
+    return (
+      <PFNotchedOutline label={this.props.intl.formatMessage(messages.photo)}>
+        <input
+          type='file'
+          ref={this.photoUploadInput}
+          onChange={(e) => this.setState({formPhotoFile: e.target.files[0]})} />
+        <Button
+          className='pf-button-primary'
+          type='button'
+          onClick={() => this.photoUploadInput.current.click()}
+          disabled={this.state.formPhotoUrl != ""}>
+          {this.props.intl.formatMessage(messages.uploadPhoto)}
+        </Button>
+        {photoFilenameSpan}
+        {photoDeleteButton}
+        <TextField
+          label={this.props.intl.formatMessage(messages.orEnterPhotoUrl)}
+          outlined
+        >
+          <Input
+            value={this.state.formPhotoUrl}
+            onChange={(e) => this.setState({formPhotoUrl: e.target.value})}
+            disabled={this.state.formPhotoFile != ""} />
+        </TextField>
+      </PFNotchedOutline>
+    );
+  }
+
+  /*
+   * Adds a profile page field for the given site to the form.
+   */
+  addProfilePageField(site) {
+    this.setState({
+      formProfilePages: this.state.formProfilePages.concat(new Map(
+          [['site', site], ['value', '']])),
+      showProfilePageOptions: false,
+    });
+  }
+
+  /*
+   * Removes the given profile page field from the form.
+   */
+  removeProfilePageField(index) {
+    var newProfileList = this.state.formProfilePages.slice(0, index);
+    if (index < this.state.formProfilePages.length - 1) {
```

Comment:
I believe this condition is not necessary because slice() safely returns an empty array if index is the last element.

Reply:
Using immutable.List now, so resolving the thread.